draft-ietf-tls-esni.md

```diff
@@ -603,6 +606,26 @@ length.
 Backend servers (as described in {{server-behavior}}) MUST support the
 "ech_is_inner" extension.
 
+### GREASE PSK {#grease-psk}
+
+When offering ECH, the client is not permitted to advertise PSK identities in
+the ClientHelloOuter. However, it is useful to allow the client to add a fake
+"pre_shared_key" extension to the ClientHelloOuter whenever it advertises PSK
+identities in the ClientHelloInner. This provides cover traffic for session
+resumption with the backend server. When resuming a session with the client, the
+backend server sends a "pre_shared_key" extension in its ServerHello. This would
+appear to a network observer as if the the server were sending this extension
+without solicitation, which would violate the extension rules described in
+{{RFC8446}}. Sending a fake "pre_shared_key" extension in the ClientHelloOuter
+makes it appear to the network as if the extension were negotiated properly.
+
+The client generates the extension payload by constructing an `OfferedPsks`
+structure (see {{RFC8446}}, Section 4.2.11) as follows. For each PSK identity
```

Comment:
As @kaduk pointed out to me recently, TLS 1.3 actually forbids sending pre_shared_keys without psk_key_exchange_modes and even mandates the server enforce this. If we're going to start patching in individual fields, we probably need to say that too.
https://tools.ietf.org/html/rfc8446#section-4.2.9

Reply:
🤦 sigh, yeah.

Reply:
Yeah, I noticed that too. I guess I sort of thought this would be implied, but being explicit about it seems like a good idea.

Reply:
The client now MUST copy "psk_key_exchange_modes" from CHInner into CHOuter.